src/TTImp/Elab/RunElab.idr

```diff
@@ -63,9 +63,14 @@ elabScript fc nest env script@(NDCon nfc nm t ar args) exp
                               !(sc defs (toClosure withAll env
                                               !(quote defs env act'))) exp
                   x => failWith defs $ "non-function RHS of a Bind: " ++ show x
-    elabCon defs "Fail" [_,msg]
+    elabCon defs "Fail" [_, mbfc, msg]
         = do msg' <- evalClosure defs msg
-             throw (GenericMsg fc ("Error during reflection: " ++
+             customFC <- reify defs !(evalClosure defs mbfc)
+             let customFC' = case the (Maybe FC) customFC of
+                               Nothing      => fc
+                               Just EmptyFC => fc
+                               Just x       => x
```

Comment:
Should we just take an `FC` then given that `Nothing` and `Just EmptyFC` are treated the same way?